tests/meta/archive/test_archiving.py

```diff
@@ -0,0 +1,338 @@
+    mock_ddn_refresh_response._content = WriteStream.write_stream_from_content(
+        content={
+            "access": new_token,
+            "expire": new_expiration.timestamp(),
+        },
+        file_format=FileFormat.JSON,
+    ).encode()
+
+    # WHEN refreshing the auth token
+    with mock.patch(
+        FUNCTION_TO_MOCK,
+        return_value=mock_ddn_refresh_response,
+    ):
+        ddn_dataflow_api._refresh_auth_token()
+
+    # THEN the api token and the expiration time should be updated
+    assert ddn_dataflow_api.auth_token == new_token
+    assert ddn_dataflow_api.token_expiration.second == new_expiration.second
+
+
+def test_archive_folders(
+    ddn_dataflow_api, local_directory, remote_path, full_remote_path, full_local_path
+):
+    """Tests that the archiving function correctly formats the input and sends API request."""
+
+    # GIVEN two paths that should be archived
+
+    mock_ddn_refresh_response = Response()
+    mock_ddn_refresh_response.status_code = 200
+
+    # WHEN running the archive method and providing two paths
+    with mock.patch.object(
+        APIRequest,
+        "api_request_from_content",
+        return_value=mock_ddn_refresh_response,
+    ) as mock_request_submitter:
+        response: bool = ddn_dataflow_api.archive_folders(
+            {Path(local_directory): Path(remote_path)}
+        )
+
+    # THEN a boolean response should be returned
+    assert response
+
+    # THEN the mocked submit function should have been called exactly once with correct arguments
+    mock_request_submitter.assert_called_once_with(
+        api_method=APIMethods.POST,
+        url=urljoin(base=ddn_dataflow_api.url, url=DataflowEndpoints.ARCHIVE_FILES),
+        headers=dict(ddn_dataflow_api.headers, **ddn_dataflow_api.auth_header),
+        json={
+            "pathInfo": [{"source": full_local_path, "destination": full_remote_path}],
+            "osType": OSTYPE,
+            "createFolder": False,
+            "metadataList": [],
+        },
+    )
+
+
+def test_retrieve_folders(
+    ddn_dataflow_api, local_directory, remote_path, full_remote_path, full_local_path
+):
+    """Tests that the retrieve function correctly formats the input and sends API request."""
+
+    # GIVEN two paths that should be retrieved
+    mock_ddn_refresh_response = Response()
+    mock_ddn_refresh_response.status_code = 200
+
+    # WHEN running the retrieve method and providing two paths
+    with mock.patch.object(
+        APIRequest,
+        "api_request_from_content",
+        return_value=mock_ddn_refresh_response,
+    ) as mock_request_submitter:
+        response: bool = ddn_dataflow_api.retrieve_folders(
+            {Path(remote_path): Path(local_directory)}
+        )
+
+    # THEN a boolean response should be returned
+    assert response
+
+    # THEN the mocked submit function should have been called exactly once with correct arguments
+    mock_request_submitter.assert_called_once_with(
```

Comment:
assert?

Reply:
The method performs an assert and has no return so this is the intended use